test/integration/datapath/datapath_linux_test.go

```diff
@@ -0,0 +1,325 @@
+	gpServiceAccountPath       = gpFolder + "/service-account.yaml"
+	gpDaemonset                = gpFolder + "/daemonset.yaml"
+	gpDaemonsetIPv6            = gpFolder + "/daemonset-ipv6.yaml"
+	gpDeployment               = gpFolder + "/deployment.yaml"
+)
+
+var (
+	podPrefix        = flag.String("podName", "goldpinger", "Prefix for test pods")
+	podNamespace     = flag.String("namespace", "default", "Namespace for test pods")
+	nodepoolSelector = flag.String("nodepoolSelector", "nodepool1", "Provides nodepool as a Node-Selector for pods")
+	testProfile      = flag.String("testName", LinuxDeployIPV4, "Linux datapath test profile")
+	defaultRetrier   = retry.Retrier{
+		Attempts: 10,
+		Delay:    defaultRetryDelaySeconds * time.Second,
+	}
+)
+
+/*
+This test assumes that you have the current credentials loaded in your default kubeconfig for a
+k8s cluster with a Linux nodepool consisting of at least 2 Linux nodes.
+*** The expected nodepool name is npwin, if the nodepool has a diferent name ensure that you change nodepoolSelector with:
+	-nodepoolSelector="yournodepoolname"
+
+To run the test use one of the following commands:
+go test -count=1 test/integration/datapath/datapath_linux_test.go -timeout 3m -tags connection -run ^TestDatapathLinux$ -tags=connection,integration
+   or
+go test -count=1 test/integration/datapath/datapath_linux_test.go -timeout 3m -tags connection -run ^TestDatapathLinux$ -podName=acnpod -nodepoolSelector=aks-pool1 -tags=connection,integration
+
+
+This test checks pod to pod, pod to node, pod to Internet check
+
+Timeout context is controled by the -timeout flag.
+
+*/
+
+// return podLabelSelector and nodeLabelSelector
+func createLabelSelectors() (string, string) {
+	return fmt.Sprintf("%s=%s", podLabelKey, *podPrefix), fmt.Sprintf("%s=%s", nodepoolKey, *nodepoolSelector)
+}
+
+func setupLinuxEnvironment(t *testing.T) {
+	ctx := context.Background()
+
+	t.Log("Create Clientset")
+	clientset, err := k8sutils.MustGetClientset()
+	if err != nil {
+		require.NoError(t, err, "could not get k8s clientset: %v", err)
+	}
+
+	t.Log("Create Label Selectors")
+	podLabelSelector, nodeLabelSelector := createLabelSelectors()
+
+	t.Log("Get Nodes")
+	nodes, err := k8sutils.GetNodeListByLabelSelector(ctx, clientset, nodeLabelSelector)
+	if err != nil {
+		require.NoError(t, err, "could not get k8s node list: %v", err)
+	}
+
+	createPodFlag := !(apierrors.IsAlreadyExists(err))
+	t.Logf("%v", createPodFlag)
+
+	if createPodFlag {
+		var daemonset appsv1.DaemonSet
+		t.Log("Creating Linux pods through deployment")
+		deployment, err := k8sutils.MustParseDeployment(*testProfile)
+		if err != nil {
+			require.NoError(t, err)
+		}
+
+		if *testProfile == LinuxDeployIPV4 {
+			daemonset, err = k8sutils.MustParseDaemonSet(gpDaemonset)
+			if err != nil {
+				t.Fatal(err)
+			}
+		} else {
+			daemonset, err = k8sutils.MustParseDaemonSet(gpDaemonsetIPv6)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
```

Comment:
nit: This can be replaced with(easier to scale if we need to add more test profiles.)
```
var gpDaemonset= map[string]string{
	"LinuxDeployIPV4":  "filename_linux",
	"DualStack"      :  "filename"   
}
...
daemonset, err = k8sutils.MustParseDaemonSet(gpDaemonset[*testProfile])
if err != nil {
	t.Fatal(err)
}
...
```